kube-core/src/request.rs

```diff
@@ -91,6 +97,16 @@ impl Request {
                 "ListParams::continue_token cannot be used with a watch.".into(),
             ));
         }
+        if lp.resource_version.is_some() {
+            return Err(Error::Validation(
+                "ListParams::resource_version cannot be used with a watch.".into(),
+            ));
+        }
+        if lp.resource_version_match.is_some() {
+            return Err(Error::Validation(
+                "ListParams::resource_version_match cannot be used with a watch.".into(),
+            ));
+        }
```

Comment:
Given this note in the docs:

> Setting the resourceVersionMatch parameter without setting resourceVersion is not valid.

It might also be worth having a validation check for:

```rust
lp.resource_version_match.is_some() && lp.resource_version.is_none()
```

Reply:
Yeah, good catch!
According to the table [here](https://kubernetes.io/docs/reference/using-api/api-concepts/#semantics-for-get-and-list)  the combination of  `resourceVersionMatch=Exact` and `resourceVersion=0` is also invalid. 

I will add validations for both 👍